### ISSUE TITLE
Indentation incorrect level for chained method with bracket on new line

### ISSUE THREAD

**Issue by ati90ati** [checkstyle/checkstyle#6210]
Hello,
https://checkstyle.org/config_misc.html#Indentation
I'm using checkstyle 8.14 and I have a problem with Indentation check.

```
/var/tmp $ cat config.xml

        <module name="Indentation">
            <property name="lineWrappingIndentation" value="8"/> <!-- default is 4 -->
        </module>

/var/tmp $ cat MyClass.java

        assertThat(result).hasSize(2)
                .extracting(
                        MyDto::getField1,
                        MyDto::getField2,
                        MyDto::getField3)
                .containsOnly(     // <- this is at level 16
                        tuple(ID_1, expectedField2, expectedField3),
                        tuple(ID_2, expectedField2, expectedField3)
                );          // <- 'method call rparen' has incorrect indentation level 16, expected level should be 8. [Indentation]     <<-- line 77

/var/tmp $ RUN_LOCALE="-Duser.language=en -Duser.country=US"
/var/tmp $ java $RUN_LOCALE -jar checkstyle-X.XX-all.jar -c config.xml YOUR_FILE.java

[ant:checkstyle] [ERROR] .../MyClass.java:77: 'method call rparen' has incorrect indentation level 16, expected level should be 8. [Indentation]
```

---------------

For chained methods I expect that line 77 should start at level 16 but the checkstyle reports that it should start at 8.
Maybe a fix can be added for this. Or a different parameter if needed. Or some parameter to disable these kind of checks (tokens) and allow only indentation for others.

NOTE: `assertThat` is from `assertj-core` library.
NOTE 2: at this moment because of this bug the Indentation check is completely unusable (because we even cannot disable this rparen check).

**Comment by rnveach** [comment]
@ati90ati Please give us full code and configuration so we can reproduce issue without problem.
See https://checkstyle.org/report_issue.html#How_to_report_a_bug.3F

**Comment by johndaniels** [comment]
Hey there. I made a github repo to demonstrate this.

https://github.com/johndaniels/checkstyle-bug

**Comment by rnveach** [comment]
Case from repo is:
````
$ cat TestClass.java
package indentation;

public class Indentation {

  public Indentation indentation(Object... args) {
    return this;
  }

  public static void main(String[] args) {
    Indentation i = new Indentation();
    i.indentation()
        .indentation(
            "a",
            "b"
        );
  }
}

$ cat TestConfig.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
    "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
    "https://checkstyle.org/dtds/configuration_1_3.dtd">

<module name = "Checker">
    <property name="charset" value="UTF-8"/>

    <property name="severity" value="error"/>

    <property name="fileExtensions" value="java, properties, xml"/>

    <module name="TreeWalker">
        <module name="Indentation">
            <property name="basicOffset" value="2"/>
            <property name="braceAdjustment" value="0"/>
            <property name="caseIndent" value="2"/>
            <property name="throwsIndent" value="4"/>
            <property name="lineWrappingIndentation" value="4"/>
            <property name="arrayInitIndent" value="2"/>
        </module>
    </module>
</module>

$ java -jar checkstyle-8.17-all.jar -c TestConfig.xml TestClass.java
Starting audit...
[ERROR] TestClass.java:15: 'method call rparen' has incorrect indentation level 8, expected level should be 6. [Indentation]
Audit done.
Checkstyle ends with 1 errors.
````

**Comment by johndaniels** [comment]
Sorry about that. Thanks for putting it in the more proper format!

**Comment by dalayach** [comment]
Just poking the thread to see if there has been any updates.

For those who just don't want to see the error (like me), well, Indentation is more a readability issue in Java than anything else. If it's going to clog up my list of errors, better to simply remove the module until the fix is released. You can do so by ctrl+f "Indentation", and removing that line and it's property name's.

**Comment by rnveach** [comment]
@dalayach No updates or activity. Anyone is free to help with a fix.

**Comment by dalayach** [comment]
I will do my best. I'm still in school, so forgive me if it takes forever.

**Comment by romani** [comment]
@dalayach , thanks a lot for desire to help, just do code changes and provide Pull request.  We will guide you further.

**Comment by alabotski** [comment]
8.21 I have the some problems

**Comment by romani** [comment]
Fix is merged.
@alinkov , thanks a lot for your help.

**Comment by vilppuvuorinen** [comment]
The fix in 8.29 works only if the first line does not contain method chaining. I know, the examples are horrible.

Following snippets use `<property name="lineWrappingIndentation" value="8"/>`.

This case is fine:

```java
Arrays.asList("a", "b", "c")
                .stream()
                .map(String::toUpperCase)
                .forEach(
                        System.out::println
                );
```

In this case we get flashbacks from 8.28:

```java
Arrays.asList("a", "b", "c").stream()
                .map(String::toUpperCase)
                .forEach(
                        System.out::println
                );
```

Accepted formatting is:

```java
Arrays.asList("a", "b", "c").stream()
                .map(String::toUpperCase)
                .forEach(
                        System.out::println
);
```

**Comment by romani** [comment]
@vilppuvuorinen, please open new issue with all details.
